apps/files/utils.py

```diff
@@ -456,13 +456,16 @@ def parse_xpi(xpi, addon=None, check=True):
 
 def check_xpi_info(xpi_info, addon=None):
     from addons.models import Addon, BlacklistedGuid
-    if not xpi_info['guid']:
+    guid = xpi_info['guid']
+    if not guid:
         raise forms.ValidationError(_("Could not find a UUID."))
-    if addon and addon.guid != xpi_info['guid']:
+    if len(guid) > 64:
+        raise forms.ValidationError(_("UUID must be less than 64chars"))
```

Comment:
"Add-on ID must be 64 characters or shorter"

The warnings above and below should also say "an add-on ID" rather than "a UUID", since most add-ons IDs aren't UUIDs.